src/main/java/org/jsoup/helper/Validate.java

```diff
@@ -9,6 +9,18 @@ public final class Validate {
     
     private Validate() {}
 
+    /**
+     * Validates that the two objects are not the same.
+     * @param a First object
+     * @param b Second object
+     * @param message Error message to use if the objects are the same
+     * @throws IllegalArgumentException if the objects are the same
+     */
+    public static void notStrictlyEqual(Object a, Object b, String message) {
```

Comment:
Personally, I would call this method something like "NotSameObject". To make it clearer that it's an identity check.